app/controllers/admin/events_controller.rb

```diff
@@ -23,11 +23,41 @@ def index
       @tracks_distribution_confirmed = @conference.tracks_distribution(:confirmed)
       @event_distribution = @conference.event_distribution
       @scheduled_event_distribution = @conference.scheduled_event_distribution
+      @file_name = "events_for_#{@conference.short_title}"
+      @my_var = params[:my_var]
 
       respond_to do |format|
         format.html
         # Explicity call #to_json to avoid the use of EventSerializer
         format.json { render json: Event.where(state: :confirmed, program: @program).to_json }
+        format.xlsx do
+          if @my_var == 'confirmed'
+            render xlsx: 'confirmed_events', filename: "#{@file_name}.xlsx"
+          elsif @my_var == 'all'
+            render xlsx: 'all_events', filename: "#{@file_name}.xlsx"
+          elsif @my_var == 'all_with_comments'
+            render xlsx: 'all_with_comments', filename: "#{@file_name}.xlsx"
+          end
+        end
+        format.pdf do
+          if @my_var == 'confirmed'
+            render 'confirmed_events'
```

Comment:
So I have been trying this again after our talk, and you can actually render 1 file for every format. Eg. 
`format.pdf { render 'events_pdf' }`

And in your file (events_pdf.pdf.prawn) you can call the individual files you want. Eg.
`render 'confirmed_events'` (which will render the partial _confirmed_events.pdf.prawn)